pkg/task/task.go

```diff
@@ -126,6 +131,49 @@ func (pr *taskRunner) RunTask(ctx context.Context, commandContext *CommandContex
 	return commandContext.OriginalError
 }
 
+// Temporary method to execute Task with a checkpoint file. Only executable from the Upgrade command
+func (pr *taskRunner) RunTaskWithCheckpoint(ctx context.Context, commandContext *CommandContext) error {
+	commandContext.Profiler = &Profiler{
+		metrics: make(map[string]map[string]time.Duration),
+		starts:  make(map[string]map[string]time.Time),
+	}
+	task := pr.task
+	start := time.Now()
+	defer taskRunnerFinalBlock(start)
+
+	checkpointFileName := fmt.Sprintf("%s-checkpoint.yaml", commandContext.ClusterSpec.Cluster.Name)
+	checkpointInfo, err := setupCheckpointInfo(commandContext, checkpointFileName)
+	if err != nil {
+		return err
+	}
+
+	for task != nil {
+		if _, ok := checkpointInfo.CompletedTasks[task.Name()]; ok {
+			nextTask, err := task.Restore(ctx, commandContext, checkpointInfo.CompletedTasks[task.Name()])
+			if err != nil {
+				return fmt.Errorf("restoring checkpoint info: %v", err)
+			}
+			task = nextTask
+			continue
+		}
+		logger.V(4).Info("Task start", "task_name", task.Name())
+		commandContext.Profiler.SetStartTask(task.Name())
+		nextTask := task.Run(ctx, commandContext)
+		commandContext.Profiler.MarkDoneTask(task.Name())
+		commandContext.Profiler.logProfileSummary(task.Name())
```

Comment:
Why do we have some methods in the Profiler as exported and some as unexported funcs?